### PR TITLE
Open with

### PR DESCRIPTION
**Pull request by jburel**
# What this PR does

Review the Open With option introduced a while back
At the time, all the images were downloaded as OME-TIFF
This PR adds the option to download the file in its original format

# Testing this PR

* Open an image using the Open With option
* Check that the original image is downloaded
* Set the option to false in the configuration file
* Check that an OME-TIFF is generated

# Related reading

https://www.openmicroscopy.org/community/viewtopic.php?f=4&t=8516

### CODE REVIEW

**Comment by dominikl** [comment]
Works like expected, good to merge 👍

**Comment by joshmoore** [comment]
Is there a substantial change in the screenshot? Any docs which need updating?

**Comment by jburel** [comment]
There are no UI changes and this feature is not indicated in ome-help (only the web option)

File: components/insight/SRC/org/openmicroscopy/shoola/env/ui/OpenObjectLoader.java [hunk 1]
```diff
@@ -93,14 +119,22 @@ public void load()
...
-    		name += "_"+image.getId();
-    		path += UIUtilities.replaceNonWordCharacters(name)+"."+OMETIFFFilter.OME_TIFF;
-    		f = new File(path);
-    		f.deleteOnExit();
-    		handle = ivView.exportImageAsOMETiff(ctx, image.getId(), f, null,
-    				this);
+    		if (originalImage) {
+    		    List<DataObject> objects = new ArrayList<DataObject>();
+    		    objects.add(image);
+    		    f = new File(folderPath);
```
Comment (joshmoore) [inline components/insight/SRC/org/openmicroscopy/shoola/env/ui/OpenObjectLoader.java:126]: Will this `folderPath` get cleaned up on exit?

Comment (jburel) [reply components/insight/SRC/org/openmicroscopy/shoola/env/ui/OpenObjectLoader.java:126]: I will need to check.
This is very old part of the code i.e. where we store things

Comment (jburel) [reply components/insight/SRC/org/openmicroscopy/shoola/env/ui/OpenObjectLoader.java:126]: I reviewed the code. The file is stored in the specified tmp directory
Nothing was deleted on exit

**Comment by joshmoore** [comment]
Steps taken:

 * downloaded from ci-master
 * set OpenWith/Original to `true`
 * logged into eel and used open_with fiji.app on an image
 * saw that it was created as $TMPDIR/test_image.png
 * closed fiji & insight: the file wasn't deleted.

Note: I also tested with a multi-file download. Both files were downloaded to $TMPDIR correctly, but were also not deleted.

**Comment by jburel** [comment]
It must have been the case since we introduce the feature since ``deleteOnExit`` was invoked at least in the fiji case

**Comment by jburel** [comment]
Which platform are you using?
The file is deleted on mac

**Comment by joshmoore** [comment]
OS X (10.12.6)

```
# joshuas-mbp:~ jamoore$ echo $TMPDIR
/var/folders/z5/txc_jj6x5l5cm81r56ck1n9c0000gn/T/
# joshuas-mbp:~ jamoore$ ls $TMPDIR/{open_with,test_import}*
/var/folders/z5/txc_jj6x5l5cm81r56ck1n9c0000gn/T//open_with.fake
/var/folders/z5/txc_jj6x5l5cm81r56ck1n9c0000gn/T//open_with.fake.ini
/var/folders/z5/txc_jj6x5l5cm81r56ck1n9c0000gn/T//test_import.fake
/var/folders/z5/txc_jj6x5l5cm81r56ck1n9c0000gn/T//test_import.png
```

**Comment by jburel** [comment]
problem should be sorted now

**Review by joshmoore — APPROVED** [review]
Files are definitely deleted now. I'd say let's go for it.

For a potential follow-up, one thing I noted: since the last files hadn't been deleted, the new download had their names changed:

```
-rw-r--r--    1 jamoore  staff          0 Jun 29 13:16 open_with.fake(1).ini
-rw-r--r--    1 jamoore  staff          0 Jun 29 13:16 open_with(1).fake
```

This strategy will be unsafe for some file formats and instead a separate subdirectory per fileset download should be used.

**Comment by jburel** [comment]
The "renaming" is something not directly related to the "open with" functionality
The same situation will happen when clicking on "download"
I went for the smallest modifications required in this PR
but certainly something to potentially look into

**Comment by joshmoore** [comment]
In the download case, is the user given the chance to chose the location? If so, would that same mechanism _easily_ be an option here?

**Comment by jburel** [comment]
The user can chose the location but a file with the same name could be there too
The download/open with and possibly other options use the same method to download. The renaming logic is implemented in one method so I think the strategy indicated above should not be difficult to implement

**Comment by sbesson** [comment]
Captured follow-up todos as https://trello.com/c/mYgWd2vx/462-insight-rfe-prevent-renaming-on-download-open-with.

**Pull request merged**